rts/Game/Camera/SpringController.cpp

```diff
@@ -258,8 +261,11 @@ float CSpringController::ZoomIn(const float3& curCamPos, const float3& newDir, c
 	const float3 wantedPos = curCamPos + cursorVec * (1.0f - scaledMode);
 
 	// figure out how far we will end up from the ground at new wanted point
-	curDist = DistanceToGround(wantedPos, dir, pos.y);
-	pos = wantedPos + dir * curDist;
+	float wantedDist = DistanceToGround(wantedPos, dir, pos.y);
+	if (wantedDist > minDist) {
```

Comment:
Maybe this could get into a situation where the -delta makes this condition not pass (e.g. very high scroll delta on single tick), but you're not close to minheight. Perhaps clamp wantedDist instead?